integreat_cms/core/management/commands/repair_tree.py

```diff
@@ -0,0 +1,88 @@
+import logging
+
+from django.core.management.base import CommandError
+
+from ....cms.models import Page
+from ..log_command import LogCommand
+
+logger = logging.getLogger(__name__)
+
+
+class Command(LogCommand):
+    """
+    Management command to repair a broken Treebeard page tree
+    """
+
+    help = "Search & replace links in the content"
+
+    def add_arguments(self, parser):
+        """
+        Define the arguments of this command
+
+        :param parser: The argument parser
+        :type parser: ~django.core.management.base.CommandParser
+        """
+        parser.add_argument(
+            "--page", help="The ID of a page belongig to the broken tree."
+        )
+        parser.add_argument(
+            "--commit",
+            action="store_true",
+            help="Whether changes should be written to the database",
+        )
+
+    # pylint: disable=arguments-differ
+    def handle(self, *args, page, commit, **options):
+        r"""
+        Try to run the command
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param page: The page ID (node) of a broken tree
+        :type page: str
+
+        :param commit: Whether changes should be written to the database
+        :type commit: bool
+
+        :param \**options: The supplied keyword options
+        :type \**options: dict
+
+        :raises ~django.core.management.base.CommandError: When the input is invalid
+        """
+        if not page:
+            raise CommandError
+        root_node = Page.objects.get(id=page).get_root()
+        self.calculcate_left_right_values(root_node, 1, commit)
+
+    def calculcate_left_right_values(self, tree_node, left, commit):
+        """
+        Recursively calculate the Left and Right value for a given node and its
+        children.
+
+        :param tree_node: A node of a MPTT tree
+        :type tree_node: ~integreat_cms.cms.models.pages.page.Page
+
+        :param left: The new left value of the node
+        :type left: int
+
+        :param commit: Whether changes should be written to the database
+        :type commit: bool
+
+        :return: The new right value of the node
+        :rtype: int
+        """
+        right = left
+
+        for child in tree_node.get_children().order_by("lft"):
+            right = self.calculcate_left_right_values(child, right + 1, commit)
+
+        right += 1
+
+        logger.warning("id=%i, lft=%s, rgt=%s", tree_node.id, left, right)
+        if commit:
```

Comment:
```suggestion
        if not valid and commit:
            if tree_node.parent:
                tree_node.tree_id = tree_node.parent.tree_id
                tree_node.depth = tree_node.parent.depth + 1
```
I suggest to also update the internal fields `tree_id` and `depth` in case they are incorrect.